src/store/index.js

```diff
@@ -50,15 +52,28 @@ export default () => {
             },
             updateWaitingForInitialSearch(state, value) {
                 state.waitingForInitialSearch = value;
+            },
+            updateUrl(state) {
+                const history = createBrowserHistory();
+                if(!state.waitingForInitialSearch){
+                    history.push({
+                        search: getSearchQueryString(state.pendingSearch),
+                    });
+                }
+            },
+            checkUrlForFacetSelection(state) {
+                state.pendingSearch = parseSearchQueryString(location.search);
             }
         },
         actions: {
             fetchResults({ commit, state }, searchParams) {
+                commit('checkUrlForFacetSelection');
```

Comment:
Parsing the URL search params is necessary is needed only on the initial search. Migrate this as a function invoked in the initialSearch method in HakseearchVue.js